<a href="https://colab.research.google.com/github/vg137/ibd-causal-multiomics/blob/main/nb2_gene_expression_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Introduction and setup

In [24]:
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [38]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import pyarrow as pa
import pyarrow.parquet as pq

In [39]:
data_dir = "/content/drive/My Drive/Colab Notebooks/Genomics project/ibd_data"
expression_subdir = "expression"
geo_series_name = "GSE57945"
expression_filepath = f"{data_dir}/{expression_subdir}/{geo_series_name}/{geo_series_name}_norm_counts_TPM_GRCh38.p13_NCBI.tsv.gz"

snp2gene_subdir = "snp2gene"
snp2gene_filepath = f"{data_dir}/{snp2gene_subdir}/GCST90292538_snp2gene.parquet"

We can load the expression matrix with the expression values of different samples

In [37]:
expr = pd.read_csv(
    expression_filepath,
    sep="\t",
    index_col=0,
)
expr.head()

,GSM1598408,GSM1598409,GSM1598410,GSM1598411,GSM1598412,GSM1598413,GSM1598414,GSM1598415,GSM1598416,GSM1598417,...,GSM1598720,GSM1598721,GSM1598722,GSM1598723,GSM1598724,GSM1598725,GSM1598726,GSM1598727,GSM1598728,GSM1598729
GeneID,,,,,,,,,,,,,,,,,,,,,
100287102,0.1633,0.2159,0.06581,0.0955,0.000,0.2063,0.282,0.355,0.2475,0.000,...,0.07788,0.0943,0.000,0.248,0.2734,0.1704,0.2731,0.1742,0.1736,0.2991
653635,7.3950,4.3020,9.34200,7.6700,3.146,11.6600,20.540,15.080,13.6300,7.360,...,18.04000,28.0900,12.810,18.600,23.2300,21.7900,35.4000,20.9800,10.9400,24.6700
102466751,3.9660,3.4970,4.79700,2.3200,1.860,5.0120,15.980,8.625,2.0040,1.896,...,13.24000,13.7500,8.768,10.040,9.9620,8.2770,15.9300,10.5800,4.2170,12.1100
107985730,0.0000,0.0000,0.00000,0.0000,0.000,0.0000,0.000,0.000,0.0000,0.000,...,0.00000,0.0000,0.000,0.000,0.4197,0.0000,0.1677,0.0000,0.0000,0.0000
100302278,0.0000,0.0000,0.00000,0.0000,0.000,0.0000,0.000,0.000,0.0000,0.000,...,0.00000,0.0000,0.000,0.000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000


Let us see how many rows and columns this dataframe has

In [34]:
expr.shape

(39376, 322)

So, we have 39376 different genes and 322 biological samples from which the RNA was sequenced

In [46]:
gene_average = expr.mean(axis=1)
gene_var = expr.var(axis=1)
gene_average/gene_var

,0
GeneID,
100287102,5.264482
653635,0.340993
102466751,0.249028
107985730,0.426572
100302278,0.672195
...,...
4541,0.000943
4556,0.000656
4519,0.000411
